docs/source-pytorch/tuning/profiler.rst

```diff
@@ -4,6 +4,29 @@
 Find bottlenecks in your code
 #############################
 
+.. warning::
+
+    **Do not wrap** ``Trainer.fit()``, ``Trainer.validate()``, or other Trainer methods
+    inside a manual ``torch.profiler.profile`` context manager.
+    This will cause unexpected crashes and cryptic errors due to incompatibility between
+    PyTorch Profiler's context management and Lightning's internal training loop.
+    Instead, always use the ``profiler`` argument in the ``Trainer`` constructor.
+
+    Example (correct usage):
+
+    .. code-block:: python
+
+        import pytorch_lightning as pl
+
+        trainer = pl.Trainer(
+            profiler="pytorch",  # <- This enables built-in profiling safely!
+            ...
+        )
+        trainer.fit(model, train_dataloaders=...)
+
+    **References:**
+      - https://github.com/pytorch/pytorch/issues/88472
+
```

Comment:
There is really not a reason for users to know the exact issue. Also even if the user looked at the issue I will say that it does not really help, because there is no mention of profiling in that so it will only lead to more confusion.
```suggestion
```